lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -1644,40 +1643,161 @@ def test_certificate_generation_for_students(self):
         for student in students[2:7]:
             CertificateWhitelistFactory.create(user=student, course_id=self.course.id, whitelist=True)
 
-        current_task = Mock()
-        current_task.update_state = Mock()
-        instructor_task = Mock()
-        instructor_task.task_input = json.dumps({'students': None})
+        task_input = {'student_set': None}
+        expected_results = {
+            'action_name': 'certificates generated',
+            'total': 10,
+            'attempted': 8,
+            'succeeded': 5,
+            'failed': 3,
+            'skipped': 2
+        }
+
         with self.assertNumQueries(214):
-            with patch('instructor_task.tasks_helper._get_current_task') as mock_current_task:
-                mock_current_task.return_value = current_task
-                with patch('capa.xqueue_interface.XQueueInterface.send_to_queue') as mock_queue:
-                    mock_queue.return_value = (0, "Successfully queued")
-                    with patch('instructor_task.models.InstructorTask.objects.get') as instructor_task_object:
-                        instructor_task_object.return_value = instructor_task
-                        result = generate_students_certificates(
-                            None, None, self.course.id, {}, 'certificates generated'
-                        )
-        self.assertDictContainsSubset(
-            {
-                'action_name': 'certificates generated',
-                'total': 10,
-                'attempted': 8,
-                'succeeded': 5,
-                'failed': 3,
-                'skipped': 2
-            },
-            result
+            self.assertCertificatesGenerated(task_input, expected_results)
+
+    def test_certificate_generation_all_whitelisted(self):
+        """
+        Verify that certificates generated for all white-listed students when using semantic task_input as
+        `all_whitelisted`.
+        """
+        # create 5 students
+        students = self._create_students(1, 6)
```

Comment:
why does this file insist on starting ranges at 1? We'll simplify these tests (and the `_create_students` method) if we just have them start at 0. Their names don't matter — only that they're distinct.